cpp/src/arrow/compute/exec/options.h

```diff
@@ -111,5 +111,32 @@ class ARROW_EXPORT SinkNodeOptions : public ExecNodeOptions {
   std::function<Future<util::optional<ExecBatch>>()>* generator;
 };
 
+enum JoinType {
```

Comment:
nit: enum class? (Though the compute functions are already somewhat inconsistent about this)

Reply:
I see.. yes, I can do that :-) Do we give preference to enum class over plain enums?

Reply:
Hmm. The C++ style guide doesn't say anything. I would say enum class is generally preferred in modern C++. We semi-intentionally use plain enum in a few places (when nested inside a struct) but I'm not sure why we have that pattern. 

This isn't the only place where this happens so it's not a big deal, but I would prefer enum class so that it's not implicitly converted and so that namespacing is explicit.